src/neo4j_graphrag/experimental/pipeline/config/config_parser.py

```diff
@@ -0,0 +1,739 @@
+        return klass  # type: ignore[no-any-return]
+
+    def resolve_param(self, param: ParamConfig) -> Any:
+        """Finds the parameter value from its definition."""
+        if not isinstance(param, ParamToResolveConfig):
+            # some parameters do not have to be resolved, real
+            # values are already provided
+            return param
+        # all ParamToResolveConfig have a resolver_ field
+        return param.resolve(self._global_data)
+
+    def resolve_params(self, params: dict[str, ParamConfig]) -> dict[str, Any]:
+        """Resolve all parameters
+
+        Returning dict[str, Any] because parameters can be anything (str, float, list, dict...)
+        """
+        return {
+            param_name: self.resolve_param(param)
+            for param_name, param in params.items()
+        }
+
+    @abc.abstractmethod
+    def parse(self, resolved_data: dict[str, Any] | None = None) -> Any:
+        raise NotImplementedError()
+
+
+T = TypeVar("T")
+"""Generic type to help (a bit) mypy with the return type of the parse method"""
+
+
+class ObjectConfig(AbstractConfig, Generic[T]):
+    """A config class to represent an object from a class name
+    and its constructor parameters.
+    """
+
+    class_: str | None = Field(default=None, validate_default=True)
+    """Path to class to be instantiated."""
+    params_: dict[str, ParamConfig] = {}
+    """Initialization parameters."""
+
+    DEFAULT_MODULE: ClassVar[str] = "."
+    """Default module to import the class from."""
+    INTERFACE: ClassVar[type] = object
+    """Constraint on the class (must be a subclass of)."""
+    REQUIRED_PARAMS: ClassVar[list[str]] = []
+    """List of required parameters for this object constructor."""
+
+    @field_validator("params_")
+    @classmethod
+    def validate_params(cls, params_: dict[str, Any]) -> dict[str, Any]:
+        """Make sure all required parameters are provided."""
+        for p in cls.REQUIRED_PARAMS:
+            if p not in params_:
+                raise ValueError(f"Missing parameter {p}")
+        return params_
+
+    def get_module(self) -> str:
+        return self.DEFAULT_MODULE
+
+    def get_interface(self) -> type:
+        return self.INTERFACE
+
+    def parse(self, resolved_data: dict[str, Any] | None = None) -> T:
+        """Import `class_`, resolve `params_` and instantiate object."""
+        self._global_data = resolved_data or {}
+        if self.class_ is None:
+            raise ValueError("`class_` is not defined")
+        klass = self._get_class(self.class_, self.get_module())
+        if not issubclass(klass, self.get_interface()):
+            raise ValueError(
+                f"Invalid class '{klass}'. Expected a subclass of '{self.get_interface()}'"
+            )
+        params = self.resolve_params(self.params_)
+        try:
+            obj = klass(**params)
+        except TypeError as e:
+            raise e
+        return obj  # type: ignore[return-value]
```

Comment:
Must we do `# type: ignore[return-value]` here?

Reply:
Yes, we have to, because we're dynamically loading a class type using `importlib`, and this returns `Any` (because we can load a function or any other variable declared in the module using the same function).

Reply:
I would generally prefer to avoid ignoring types unless it is in some test files or special cases. Is there a way around this, maybe combining a runtime check with `typing.cast`?

```python
T = TypeVar("T", bound=object)

...

if not issubclass(obj, self.get_interface()):
        raise TypeError(...)
return cast(T, obj)
```

Reply:
Ok, done something along what you propose in https://github.com/neo4j/neo4j-graphrag-python/pull/220/commits/ce21353a4b81e8315f1b77070d28c049b97572c1 (and also in https://github.com/neo4j/neo4j-graphrag-python/pull/220/commits/3122edaf35d91bbfdd8f3eed4ad67066973a5fba for `_get_class`)

The type check with `get_interface` is performed above with `issubclass`.